pinecone/client.go

```diff
@@ -59,6 +143,49 @@ func NewClient(in NewClientParams) (*Client, error) {
 	return NewClientBase(NewClientBaseParams{Headers: clientHeaders, Host: in.Host, RestClient: in.RestClient, SourceTag: in.SourceTag})
 }
 
+// NewClientBase creates and initializes a new instance of Client with custom authentication headers,
+// allowing users to authenticate in ways other than passing an API key.
+//
+// This function requires an input parameter of type NewClientBaseParams, which includes:
+//   - Headers: A map of additional HTTP headers to include in the API request.
+//     "Authorization" and "X-Project-Id" headers are required.
+//   - Host: The host URL of the Pinecone control plane API. If not provided,
+//     the default value is "https://api.pinecone.io".
+//   - RestClient: An optional custom HTTP client to use for communication with the control plane API.
+//   - SourceTag: An optional string used to help Pinecone attribute API activity to our partners.
+//
+// Returns a pointer to an initialized Client instance on success. In case of
+// failure, it returns nil and an error describing the issue encountered. Possible errors
+// include issues with setting up the headers or problems initializing the
+// underlying REST client.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientBaseParams{
+//    Headers: map[string]string{
+//     "Authorization": "Bearer " + "<your JWT token>"
+//     "X-Project-Id": "<Your Pinecone project ID>"
+//      },
```

Comment:
Ah, I see where the `X-Project-Id` came from. Yeah I'm not sure how specific we should be in these doc comments around this stuff. This is mostly for us to integrate with things internally at the moment, but users could use it more generally. We should look at getting @jhamon's take on this if possible. 

We could maybe just show the `Api-Key` example here which is a little simple but demonstrates the principle.

Reply:
We couldddd but I do think having a nice code snippet to c&p for when ppl are using JWT auth is super nice. Agree let's hear from @jhamon !